### DISCUSSION TITLE
Problem with inverse search

### DISCUSSION THREAD

**Discussion by TinchoteSK** [sumatrapdfreader/sumatrapdf#4311] [Q&A]
I'm having this issue with inverse search. I'm creating a large document that is made out of about 10 .tex files. Inverse search works fine with all files but one. When I double click in Sumatra over text that came from that specific file, the editor reports a broken command line. Concretely, the file name that Sumatra provides the editor looks like "c:\users\myuser\docum" and nothing else. 

The problem does not arise with another viewers (so it's not an obvious problem with the synctex file), and does arise with different editors (I tried "notepad %f" as the inverse search string in Sumatra and notepad reported the above truncated file name). Happens with both Sumatra 3.4.6 64-bit and 3.5.2 64-bit.  I also looked into the synctex file and the name of the affected file appears normal and does not show any obvious difference with the appearance of the other file names. 

I cannot submit a bug report because I don't know how to reproduce the problem, it seems to happen with this particular synctex file. I was hoping someone who knows about the internals of this may help me diagnose the issue.

**Comment by kjk** [comment]
What is your inverse search setting? Most likely the inverse search command line you provided to SumatraPDF didn't put the file name placeholder within quotes `""` which leads to file name truncation.

**Reply by TinchoteSK** [reply]
I tried the search string "notepad %f". And the truncation happens mid word, and there are no spaces in the path. 

In any case, the inverse search DOES work for other parts of the same pdf. It fails for just one of the source .tex files.

**Comment by GitHubRulesOK** [comment]
Hmmm never seen such a problem over months of historic SumatraPDF calling
There are 2 variants (perhaps 16 means in total) of `.synctex` so the variable compressed `.synctex.gz` and the uncompressed variant `.synctex` 

However their internal filenames can be Mac Linux or Windows drive mount points, and then cause problems.
I am thinking this may be a decompression issue but need to see the synctex and pdf to test the way the reverse call is built

**Reply by TinchoteSK** [reply]
I tried the "synctex edit" command on my pdf, and it outputs the full correct path instead of the truncated one that Sumatra sends.

**Reply by GitHubRulesOK** [reply]
perhaps post a copy of the settings.txt file along with the pdf and synctex (you can put all 3 in a zip but we cannot "SEE" what you are testing without those files) there are not enough details in just your comments to emulate the issue.

If it is working for parts of the synctex but not other parts that is unusual so need to consider why the synctex is causing the issue with the settings.txt and pdf. 

PDF click looks at synctex and reads then writes the found position data to command with the settings text value so if opening other files works it is not likely to be setting UNLESS the syctex and command are  a buffer issue.

Possible tests are shorten paths (always ensure there is minimal nesting) as ALL TEX AND PDF AND SyncTeX should be in ONE folder.
Also try put **portable 32 bit SumatraPDF 3.2 or 3.12 version** and the source settings.txt in same folder as PDF then drop PDF onto SumatraPDF.EXE that will invoke much earlier decompression system

**Reply by TinchoteSK** [reply]
32-bit version 3.1.2 produces the same behaviour: inverse search works fine on parts of the document that correspond to other tex files, and sends the truncated file name to the editor when doing inverse search on parts of the pdf that correspond to this particular tex file. 

I cannot share the files publicly as I'm not willing to doxx myself nor I want the document freely available. I don't mind sharing privately if there is a way.

**Reply by GitHubRulesOK** [reply]
IF 3.1.2 has same problem that suggests a rogue synctex although there are newer variants than the source that SumatraPDF uses.

It again is odd that some command filenames work and others dont, so provide all the set

ensure PDF and syncetx are unedited with exactly same compile date/time and include your SumatraPDF-settings.txt so settings can also be considered as part of that mix with a note as to which trigger point in the PDF is clicked.
 
You can send the set of files to [kkowalczyk@gmail.com](mailto:kkowalczyk@gmail.com) the developer to see if he can see why

**Comment by kjk** [comment]
It's not about the files. 

What is the `-inverse-search` and `-forward-search` cmd-line argument you're using (https://www.sumatrapdfreader.org/docs/LaTeX-integration)?

**Reply by TinchoteSK** [reply]
The inverse search command I'm using is "C:\Program Files (x86)\WinEdt\WinEdt.exe" "[Open(|%f|);SelPar(%l,8)]". But if I replace it with "notepad %f" the error occurs exactly the same, notepad displays an error message with the same truncated path.

**Reply by kjk** [reply]
You have to use `"%f"`, not just `%f` (should fix notepad as well)

**Reply by TinchoteSK** [reply]
No, that makes no difference. The string is now "C:\Program Files (x86)\WinEdt\WinEdt.exe" "[Open(|"%f"|);SelPar(%l,8)]" and I get the same error: in the parts of the pdf that correspond to other tex files the inverse search works perfectly, and on the parts that correspond to this particular tex file the truncated error still appears.

**Reply by GitHubRulesOK** [reply]
@TinchoteSK  I understand reluctance to publically show personal paths and data however some areas where I have seen fails dont seem to fit your descriptions. lets see if any fit your usage.

One area is where synctex is using different drive roots but that would not explain "same pdf works and does not work".
One  (never resolved) was the user name had strange foreign characters , again that "failed in all calls not just some".
Many failed reverse calls are when tex are in sub folders as then paths are different for each reverse call, but again you say some work and some dont thus we would need to see the running time filenames as emitted by the synctex., at that click.

Best I can suggest is try to "capture" the "%f" and compare with the location in the synctex then try to emulate a sample that also fails for submission.

so if I click a capture it shows the correct expected location
```
"C:\Users\lez\Downloads\Apps\PDF\LATeX and Reverse-Search\sync.tex" 10  
```
to get that output I use my computer username (actually my dead brother in law account !)

so to get that confirmation in options I used 
```
c:\windows\system32\cmd.exe /r "echo "%f" %l >c:\users\lez\output.txt &pause"
```
That runs echos to this user area with a pause to ensure no fails and the "%f" and %l  are the raw synctex output for that part of that PDF but note they are adapted relative not the compiled locations as my BroInLaw did not build the file !

**Reply by TinchoteSK** [reply]
There are not subfolders or different folders involved in this, and every character in the path is ascii. All tex, pdf, log, aux, synctex, etc. files are in the same folder. The offending filename does not have the longest path (longer tex files in the same pdf document work correctly with inverse search)

I did two things to  "capture the %f". 

First I replaced the inverse search string with "notepad %f". When I double click on Sumatra somewhere on the pdf, if the part of the pdf corresponds to the offending tex file notepad shows that message "file not found c:\users\myuser\docum", which tells me that's the %f. When I click on parts of the pdf corresponding to the other tex files, notepad opens the correct file. 

The second thing I did was to run "synctex edit". This is the result, which points to nothing wrong with the synctex file: 

>C:\Users\myuser>synctex edit -o 432:200:100:c:\users\myuser\documents\myfile.pdf
This is SyncTeX command line utility, version 1.5
SyncTeX result begin
Output:c:\users\myuser\documents\myfile.pdf
Input:c:\users\myuser\documents\OffendingFile.tex
Line:1575
Column:-1
Offset:0
Context:
SyncTeX result end

The third thing I did is to try with a different viewer. I used WinEdt's internal viewer, and the error does not appear. 

I also tried renaming the offending file: no change. 

Finally, I also tried starting from scratch. I copied by tex files to a different folder, compiled there, and exactly the same error appears. 

**Edit:** as a last resort, I tried emptying the offending file. And that solves the problem. So somehow this is related to the contents of the tex file. It's a long file, so it will take me a while to isolate the cause of the problem, but I'll update when I find the culprit.

**Reply by GitHubRulesOK** [reply]
I have never used `synctex edit` but you may gain more insight by simply look inside the .synctex file as text hence the request.

Just checked the MAN and as I thought there are many options for synctex looks like 31 ? from `pdftex -synctex=15 foo.tex` down to -15 thus 0=none and + or - affect textual or GZ variations.

Again it would be odd for that to make the difference  but they can be included inside each TEX file and there have an impact, AGAIN no reason for culling part of a filename !

I have to thus agree SOMETHING is affecting the synctex handler and is seemingly unique to that one .TeX

**Reply by GitHubRulesOK** [reply]
I wonder about that LAST point 
IF the TeX file itself contains a rogue unguarded syntex command word  it could be creating interference ?

For example
>TEX primitive that you can set to 1 for SyncTEX activation from the source file: \synctex=1. It can be used, for example,
to temporarily disable SyncTEX operations for some input file by properly using \synctex=0. This primitive has no effect if the -synctex=0 command line is used

Another key phrase may be `\special`  in the mix but that was possibly older TeX reserved word

**Reply by TinchoteSK** [reply]
I'm slowly adding and removing parts of the file to find the culprit. 

I want to mention that I did check the contents of the synctex file and I didn't noticed anything particular about the offending file; its name appears twice, once with .aux and once with .tex, with the correct path and it looked exactly like the other tex files in the same synctex file. 

There's also the thing that at least one other viewer (WinEdt's) and the synctex command line seem to extract the path correctly from the synctex file. 

**Edit:** well, my investigations have confusing results. I started removing and readding text to find when the inverse search for the chapter would work and when it wouldn't. The two times that I got to isolate a single paragraph the behaviour changed and the error reappeared with the suspicious paragraph removed. Then I thought that this maybe has to do with ascii/non-ascii, just because. This chapter has repeated instances of the Polish name Pełczyński. I replaced all instances of the name with a macro that I defined somewhere else. And the error disappeared. 

So, here's a conjecture (which maybe makes no sense). Could it be that Sumatra is thrown for a loop by the appearance in the synctex file of non-ascii characters? (I know that the synctex file only contains coordinates, so most likely what I'm saying makes no sense). That would explain that when the foreign characters were on a different position the error didn't appear. Anyway, I seem to have found a workaround, and I still don't know how to consistently recreate the error in my setup.
